src/components/TasksTable/index.jsx

```diff
@@ -63,16 +61,32 @@ export default class TasksTable extends Component {
     }
   );
 
-  handleHeaderClick = sortBy => {
-    const toggled = this.state.sortDirection === 'desc' ? 'asc' : 'desc';
-    const sortDirection = this.state.sortBy === sortBy ? toggled : 'desc';
+  getSortData() {
+    const { location } = this.props;
+    const query = parse(location.search.slice(1));
+
+    return {
+      sortBy: query.sortBy ? query.sortBy : 'Last Updated',
+      sortDirection: query.sortDirection ? query.sortDirection : 'desc',
+    };
+  }
+
+  handleHeaderClick = nextSortBy => {
+    const { sortBy, sortDirection } = this.getSortData();
+    const toggled = sortDirection === 'desc' ? 'asc' : 'desc';
+    const nextSortDirection = sortBy === nextSortBy ? toggled : 'desc';
 
-    this.setState({ sortBy, sortDirection });
+    this.props.history.push({
+      search: `?${stringify({
+        sortBy: nextSortBy,
+        sortDirection: nextSortDirection,
+      })}`,
+    });
   };
```

Comment:
Instead of using `nextSortDirection` and `nextSortBy`, we can make it more readable with something like:

```jsx
handleHeaderClick = sortBy => {
  const query = this.getQuery();
  const toggled = query.sortDirection === 'desc' ? 'asc' : 'desc';
  const sortDirection = query.sortBy === sortBy ? toggled : 'desc';

  this.props.history.push({
    search: `?${stringify({
      sortBy,
      sortDirection,
    })}`,
  });
};
```